File: sonar-server/src/main/webapp/WEB-INF/app/models/sonar/rule_priority.rb
#
# SonarQube, open source software quality management tool.
# Copyright (C) 2008-2014 SonarSource
# mailto:contact AT sonarsource DOT com
#
# SonarQube is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 3 of the License, or (at your option) any later version.
#
# SonarQube is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# License along with {library}; if not, write to the Free Software
# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#

class Sonar::RulePriority

  INFO = Java::OrgSonarApiRules::RulePriority::INFO
  MINOR = Java::OrgSonarApiRules::RulePriority::MINOR
  MAJOR = Java::OrgSonarApiRules::RulePriority::MAJOR
  CRITICAL = Java::OrgSonarApiRules::RulePriority::CRITICAL
  BLOCKER = Java::OrgSonarApiRules::RulePriority::BLOCKER

  PRIORITY_INFO = Java::OrgSonarApiRules::RulePriority::INFO.ordinal()
  PRIORITY_MINOR = Java::OrgSonarApiRules::RulePriority::MINOR.ordinal()
  PRIORITY_MAJOR = Java::OrgSonarApiRules::RulePriority::MAJOR.ordinal()
  PRIORITY_CRITICAL = Java::OrgSonarApiRules::RulePriority::CRITICAL.ordinal()
  PRIORITY_BLOCKER = Java::OrgSonarApiRules::RulePriority::BLOCKER.ordinal()

  def self.to_s(failure_level, translate=false)
    text = case failure_level
      when Java::OrgSonarApiRules::RulePriority::BLOCKER.ordinal()
        Java::OrgSonarApiRules::RulePriority::BLOCKER.to_s
      when Java::OrgSonarApiRules::RulePriority::CRITICAL.ordinal()
        Java::OrgSonarApiRules::RulePriority::CRITICAL.to_s
      when Java::OrgSonarApiRules::RulePriority::MAJOR.ordinal()
        Java::OrgSonarApiRules::RulePriority::MAJOR.to_s
      when Java::OrgSonarApiRules::RulePriority::MINOR.ordinal()
        Java::OrgSonarApiRules::RulePriority::MINOR.to_s
      when Java::OrgSonarApiRules::RulePriority::INFO.ordinal()
        Java::OrgSonarApiRules::RulePriority::INFO.to_s
    end

    return text unless translate
    
    i18n_key = 'severity.' + text
    result = Api::Utils.message(i18n_key, :default => as_text_map[text])
    result
  end
  
  def self.id(priority)
    begin
      javaPriority=Java::OrgSonarApiRules::RulePriority.valueOf(priority)
      javaPriority && javaPriority.ordinal
    rescue
      nil
    end
  end

  def self.as_array
    @@priorities_a ||= []
    return @@priorities_a if @@priorities_a.size > 0
    @@priorities_a << PRIORITY_INFO
    @@priorities_a << PRIORITY_MINOR
    @@priorities_a << PRIORITY_MAJOR
    @@priorities_a << PRIORITY_CRITICAL
    @@priorities_a << PRIORITY_BLOCKER
  end
  
  def self.info?(priority)
    priority==PRIORITY_INFO
  end

  def self.minor?(priority)
    priority==PRIORITY_MINOR
  end
  
  def self.major?(priority)
    priority==PRIORITY_MAJOR
  end
  
  def self.critical?(priority)
    priority==PRIORITY_CRITICAL
  end
  
  def self.blocker?(priority)
    priority==PRIORITY_BLOCKER
  end
  
  @@priority_text_map={}
  
  def self.as_text_map
    return @@priority_text_map if @@priority_text_map.size > 0
    @@priority_text_map[to_s(PRIORITY_INFO)]='Info'
    @@priority_text_map[to_s(PRIORITY_MINOR)]='Minor'
    @@priority_text_map[to_s(PRIORITY_MAJOR)]='Major'
    @@priority_text_map[to_s(PRIORITY_CRITICAL)]='Critical'
    @@priority_text_map[to_s(PRIORITY_BLOCKER)]='Blocker'
    @@priority_text_map    
  end
  
  def self.as_options
    @@priority_options ||= []
    return @@priority_options if @@priority_options.size > 0
    @@priority_options << [as_text_map[to_s(PRIORITY_INFO)], to_s(PRIORITY_INFO)]
    @@priority_options << [as_text_map[to_s(PRIORITY_MINOR)], to_s(PRIORITY_MINOR)]
    @@priority_options << [as_text_map[to_s(PRIORITY_MAJOR)], to_s(PRIORITY_MAJOR)]
    @@priority_options << [as_text_map[to_s(PRIORITY_CRITICAL)], to_s(PRIORITY_CRITICAL)]
    @@priority_options << [as_text_map[to_s(PRIORITY_BLOCKER)], to_s(PRIORITY_BLOCKER)]
    @@priority_options    
  end
end

File: sonar-plugin-api/src/main/java/org/sonar/api/rules/RulePriority.java
/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.api.rules;

import java.util.Locale;
import org.sonar.check.Priority;

/**
 * @deprecated since 4.2
 * @see org.sonar.api.rule.Severity
 */
@Deprecated
public enum RulePriority {

  /**
   * WARNING : DO NOT CHANGE THE ENUMERATION ORDER
   * the enum ordinal is used for db persistence
   */
  INFO, MINOR, MAJOR, CRITICAL, BLOCKER;

  private static final String UNKNOWN_PRIORITY = "Unknown priority ";

  /**
   * A class to map priority level prior to Sonar 1.10 to the new ones
   *
   * @param level an old priority level : Error or Warning
   * @return the corresponding RulePriority
   * @deprecated in 3.6
   */
  @Deprecated
  public static RulePriority valueOfString(String level) {
    try {
      return RulePriority.valueOf(level.toUpperCase(Locale.ENGLISH));

    } catch (IllegalArgumentException ex) {
      // backward compatibility
      if ("ERROR".equalsIgnoreCase(level)) {
        return RulePriority.MAJOR;
      } else if ("WARNING".equalsIgnoreCase(level)) {
        return RulePriority.INFO;
      }
    }
    throw new IllegalArgumentException(UNKNOWN_PRIORITY + level);
  }


  public static RulePriority fromCheckPriority(Priority checkPriority) {
    if (checkPriority == Priority.BLOCKER) {
      return RulePriority.BLOCKER;
    }
    if (checkPriority == Priority.CRITICAL) {
      return RulePriority.CRITICAL;
    }
    if (checkPriority == Priority.MAJOR) {
      return RulePriority.MAJOR;
    }
    if (checkPriority == Priority.MINOR) {
      return RulePriority.MINOR;
    }
    if (checkPriority == Priority.INFO) {
      return RulePriority.INFO;
    }
    throw new IllegalArgumentException(UNKNOWN_PRIORITY + checkPriority);
  }

  public static RulePriority valueOfInt(int ordinal) {
    return RulePriority.values()[ordinal];
  }
}
